platform/android/java/src/org/godotengine/godot/Godot.java

```diff
@@ -191,6 +191,7 @@ protected void onMainActivityResult(int requestCode, int resultCode, Intent data
 		protected void onMainPause() {}
 		protected void onMainResume() {}
 		protected void onMainDestroy() {}
+		protected boolean onMainBackPressed() { return false; }
```

Comment:
I don't get this, there's no interface to set it to true, so the added code does nothing. Also the name is confusing for a boolean, other `onMainAction()` are void.

Reply:
(e.g) handling the back key

```
        protected boolean onMainBackPressed() {
                if (moPubView.getVisibility() == View.INVISIBLE) {
                        show_banner_ad(false);
                        return true; // Handled so, godot wont quit. #742
                }
                return false;
        }
```

Reply:
if `boolean` seems odd we can do this

```
public static boolean back_key_handled = false
@Override public void onBackPressed() {
                back_key_handled = false
		for(int i=0; i<singleton_count; i++) {
                   singletons[i].onMainBackPressed())
		}
 
		System.out.printf("** BACK REQUEST!\n");
	        if (!back_key_handled) {
			GodotLib.quit();
		}
 	}
```
and we can set the back_key_handled inside the function
```
        protected void onMainBackPressed() {
                if (moPubView.getVisibility() == View.INVISIBLE) {
                        show_banner_ad(false);
                        (Godot)activity.back_key_handled = true;
                }
        }
```

Reply:
I have no strong opinion (or more accurately, no clue about Android and Java stuff :p), so calling @volzhs and @RandomShaper for review.

Reply:
current PR seems fine to me.
the alternative way looks odd to me.

Reply:
Exactly,